rudder-core/src/main/scala/com/normation/rudder/services/reports/ExecutionBatch.scala

```diff
@@ -282,15 +322,183 @@ object ExecutionBatch extends Loggable {
   )
 
   /*
-   * Utility method to factor out common logging task and be assured that
-   * the log message is actually sync with the info type.
+   * How long time a run is valid AFTER AN UPDATE (i.e, not in permanent regime).
+   * This is shorter than runValidityTime, because a config update IS a change and
+   * force to send reports in all case.
+   */
+  def updateValidityDuration(runIntervalInfo: ResolvedAgentRunInterval) = runIntervalInfo.interval.plus(GRACE_TIME_PENDING)
+
+  /*
+   * How long time a run is valid before receiving any report (but not after an update)
    */
-  private[this] def runType(traceMessage: String, runType: RunAndConfigInfo)(implicit nodeId: NodeId): RunAndConfigInfo = {
-    val msg = if(traceMessage.trim.size == 0) "" else ": " + traceMessage
-    ComplianceDebugLogger.node(nodeId).trace(s"Run config for node ${nodeId.value}: ${runType.logName} ${msg}")
-    runType
+  def runValidityDuration(runIntervalInfo: ResolvedAgentRunInterval, complianceMode: ComplianceMode) = complianceMode.mode match {
+    case ChangesOnly =>
+      //expires after run*heartbeat period - we need an other run before that.
```

Comment:
actually, it's a bit more complex than that:
- if runInterval did change, then it's indeed run*heartbeat
- else if run interval changed, we need to compute based on last run time, and expected changes in between
what is good is that we can't change the frequency when webinterface is down, so it's never during a catchup that we have changing frequencies

Reply:
I think it's OK, because `ResolvedAgentRunInterval` take into account the real run interval (i.e, the one corresponding to the run config id): if the run interval changed, we need to wait for a run which knows about the change to take it into account.